File: scripts/test_ws.py
import asyncio
import websockets

async def test():
    try:
        async with websockets.connect("ws://127.0.0.1:9999/qlcplusWS") as ws:
            print("Connected to QLC+ WebSocket!")
            await ws.send("QLC+API|getWidgetStatus|53")
            res = await asyncio.wait_for(ws.recv(), timeout=2.0)
            print("Response:", res)
    except Exception as e:
        print("Failed:", e)

asyncio.run(test())

File: scripts/test_ws.js
const WebSocket = require('ws');
const ws = new WebSocket('ws://127.0.0.1:9999/qlcplusWS');

ws.on('open', function open() {
  console.log('Connected!');
  ws.send('QLC+API|getWidgetStatus|53');
  setTimeout(() => ws.close(), 3000);
});

ws.on('message', function incoming(data) {
  console.log('Received:', data.toString());
});

ws.on('error', function error(err) {
  console.log('Error:', err);
});
